docs/codeql/codeql-language-guides/extensible-predicates.rst

```diff
@@ -0,0 +1,183 @@
+.. _extensible-predicates:
+
+:orphan:
+:nosearch:
+
+Extensible predicates and their interaction with data extensions
+================================================================
+
+You can use data extensions to model the methods and callables that control data flow in any framework or library. This is especially useful for custom frameworks or niche libraries, that are not supported by the standard CodeQL libraries.
+
+.. include:: ../reusables/beta-note-model-packs-java.rst
+
+About this article
+------------------
+
+This reference article describes the available inputs for the extensible predicates, including access paths, kinds, and provenance.
+
+Sources, sinks, summaries, and neutrals are commonly known as models. These models support several shared arguments and a few model-specific arguments. The arguments populate a series of columns for each extensible predicate.
+
+About extensible predicates
+---------------------------
+
+At a high level, there are two main components to using data extensions. The query writer defines one or more extensible predicates in their query libraries. CLI and code scanning users who want to augment these predicates supply one or more extension files whose data gets injected into the extensible predicate during evaluation. The extension files are either stored directly in the repository where the code base to be analyzed is hosted, or downloaded as CodeQL model packs.
+
+This example of an extensible predicate for a source is taken from the core Java libraries https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/dataflow/ExternalFlowExtensions.qll#L8-L11
+
+.. code-block:: ql
+
+    extensible predicate sourceModel(
+      string package, string type, boolean subtypes, string name, 
+      string signature, string ext, string output, string kind, 
+      string provenance
+    );
+
+An extensible predicate is a CodeQL predicate with the following restrictions:
+
+- It uses the ``extensible`` keyword.
+- It has no body.
+- All predicate parameters have primitive types.
+- It is not in a module.
+
+Columns shared by all extensible predicates
+-------------------------------------------
+
+The semantics of many of the columns of the extensible predicates are shared. The columns ``package``, ``type``, ``subtypes``, ``name``, and ``signature`` define which element(s) the model applies to.
+
+- ``package``: Name of the package containing the element(s) to be modeled.
+- ``type``: Name of the type containing the element(s) to be modeled.
+- ``subtypes``: A boolean flag indicating whether the model should also apply to all overrides of the selected element(s).
```

Comment:
```suggestion
- ``subtypes``: A Boolean flag indicating whether the model should also apply to all overrides of the selected element(s).
```

Reply:
This is something that I always find difficult, but I think here the lower case might be correct: https://programming.guide/java/boolean-vs-boolean.html.